langgraph_bot/tools/tools.py

```
@@ -27,3 +35,131 @@ def title_tool(posttitle:str,postdata:str):
+def python_executor(code: str) -> Dict:
+    """
+    Securely execute Python code and return structured output.
+    """
+    with tempfile.NamedTemporaryFile(
+        suffix=".py", mode="w", delete=False
+    ) as f:
+        f.write(code)
+        file_path = f.name
+
+    try:
+        process = subprocess.run(
```

Ruff (0.14.10)

`subprocess` call: check for execution of untrusted input

(S603)